Issue 4169 - UI - Port plugin tables to PF4

mreynolds389 opened the pull request:
Description:  

port the plugins tables to PF4.  This completes the entire table migration.

relates: https://github.com/389ds/389-ds-base/issues/4169

jchapma commented on src/cockpit/389-console/src/lib/plugins/dna.jsx:
```diff
@@ -1249,9 +1250,11 @@ class DNA extends React.Component {
                             <div className="ds-margin-top-xlg">
                                 <DNASharedTable
                                     rows={sharedConfigRows}
+                                    key={sharedConfigRows}
```
This is more of a general question, can we use an array as a key, my understanding is that sharedConfigRows can be modified by the user so it changes. Or am i not getting something ?

mreynolds389 replied:
So, the key can be used to tell the component that it needs to be re-rendered.  Without the key if sharedCofigRows changed it would not be reflected in the UI/component.  So when the key changes from the last render. then it tells the component to render from scratch again.

Arrays work as keys but objects do not.  So you will see places where I use a index counter as the key (tableKey) instead of an array or other variable object.

jchapma replied:
> So, the key can be used to tell the component that it needs to be re-rendered. Without the key if sharedCofigRows changed it would not be reflected in the UI/component. So when the key changes from the last render. then it tells the component to render from scratch again.
> 
> Arrays work as keys but objects do not. So you will see places where I use a index counter as the key (tableKey) instead of an array or other variable object.

Ahh I see, that's handy. Thanks for the explanation.

jchapma commented on src/cockpit/389-console/src/lib/plugins/passthroughAuthentication.jsx:
```diff
@@ -71,6 +80,120 @@ class PassthroughAuthentication extends React.Component {
 
+        this.onExcludeToggle = isExcludedOpen => {
+            this.setState({
+                isExcludedOpen
```
This is just nitpicking so ignore it if you like. Sometimes we use key: value and sometimes we use the ES6 way.

mreynolds389 replied:
This code above comes directly from patternfly 4 documentation.  We mix and match this styling up all over the code.  Do you think its something that needs to be fixed?

jchapma replied:
> This code above comes directly from patternfly 4 documentation. We mix and match this styling up all over the code. Do you think its something that needs to be fixed?

Ye i noticed the different styling, no i think its fine as it is.

jchapma replied:
LGTM

jchapma reviewed (commented):
This looks great, big improvement. A few trivial comments from me.

droideck reviewed (changes requested):
I'm not sure why you've committed the Cargo.lock change here...

The rest looks great :)
It's really a vast amount of work!

mreynolds389 committed:
Issue 4169 - UI - Port plugin tables to PF4

Description:  port the plugins tables to PF4.  This completes the entire
              table migration.

relates: https://github.com/389ds/389-ds-base/issues/4169

Reviewed by: jchapman & spichugi(Thanks!!)
